modules/javafx.base/src/main/java/javafx/collections/FXCollections.java

```diff
@@ -707,7 +706,7 @@ public static <T> void sort(ObservableList<T> list, Comparator<? super T> c) {
 
     private static class EmptyObservableList<E> extends AbstractList<E> implements ObservableList<E> {
 
-        private static final ListIterator iterator = new ListIterator() {
+        private static final ListIterator<Object> iterator = new ListIterator<>() {
```

Comment:
I considered all that, and don't mind changing it if we're in agreement.

The reason I didn't is that it would subtly change the iterator (it would have an unnecessary reference to its containing class).

Reply:
In my opinion that's fine. The iterator might not use its reference to the list, but it's using its type information.

As for creating a new iterator on every invocation (like empty set does) or returning the same instance (like empty list does), I don't mind that much and I can't imagine it mattering in terms of performance.